lib/ansible/executor/task_executor.py

```diff
@@ -41,15 +41,29 @@ def remove_omit(task_args, omit_token):
     Remove args with a value equal to the ``omit_token`` recursively
     to align with now having suboptions in the argument_spec
     '''
-    new_args = {}
 
-    for i in iteritems(task_args):
+    if isinstance(task_args, dict):
+        loop = iteritems(task_args)
+        new_args = {}
+    elif isinstance(task_args, list):
+        loop = enumerate(task_args)
+        new_args = []
+    else:
+        raise TypeError('unsupported type (%r)' % type(task_args))
+
+    for i in loop:
         if i[1] == omit_token:
             continue
-        elif isinstance(i[1], dict):
-            new_args[i[0]] = remove_omit(i[1], omit_token)
+        elif isinstance(i[1], (dict, list)):
```

Comment:
it would look cleaner this way:
```python
for i in loop:
    ind, val = i
    if val == omit_token:
        continue

    if isinstance(val, (dict, list)):
        val = remove_omit(val, omit_token)

    try:
        new_args[ind] = val
    except IndexError:
        new_args.append(val)
```

Reply:
Even better:
```python
for i in loop:
    ind, val = i
    if val == omit_token:
        continue

    if isinstance(val, (dict, list)):
        val = remove_omit(val, omit_token)

    new_args = [val] + new_args[1:]
```

Reply:
Oh, and it's `KeyError` for dict.

Reply:
I prefer the except handling over list concatenation.  The one thing this doesn't handle well is whether `new_args` is a dict vs a list.  The 2nd recommendation assumes a list.

Where possible I have a habit of not reassigning variables, as to not cause unnecessary variable re-assignment and memory shuffling.

So while cleaner or less code, it has the potential of being less performant in larger loops, and with older python versions can cause gc issues.

However, I'll take a look at it to see if there is some merger of implementations that works well.

Reply:
I'll never hit `KeyError` doing an assignment, but will hit an `IndexError` if the list is 0 length, and trying to perform a direct element assignment.